### PR TITLE
update for Reference -> Index renaming

### PR DESCRIPTION
**Pull request by blegat**

### CODE REVIEW

File: src/solverinterface.jl [hunk 1]
```diff
@@ -10,7 +10,7 @@
 Transfer the constraints of type `F`-in-`S` to the solver instance.
 """
 function copyconstraints!(m::Model, ::Type{F}, ::Type{S}) where {F<:MOI.AbstractFunction, S<:MOI.AbstractSet}
-    for cref in MOI.get(m.instance, MOI.ListOfConstraintReferences{F, S}())
+    for cref in MOI.get(m.instance, MOI.ListOfConstraintIndices{F, S}())
```
Comment (mlubin) [inline src/solverinterface.jl]: `cref` should be renamed to `cindex`

File: src/solverinterface.jl [hunk 2]
```diff
@@ -10,7 +10,7 @@
...
         f = MOI.get(m.instance, MOI.ConstraintFunction(), cref)
         s = MOI.get(m.instance, MOI.ConstraintSet(), cref)
         solvercref = MOI.addconstraint!(m.solverinstance, f, s)
```
Comment (mlubin) [inline src/solverinterface.jl]: `solvercindex`

File: src/JuMP.jl [hunk 3]
```diff
@@ -66,8 +66,8 @@ export
 
 include("utils.jl")
 
-const MOIVAR = MOI.VariableReference
-const MOICON{F,S} = MOI.ConstraintReference{F,S}
+const MOIVAR = MOI.VariableIndex
+const MOICON{F,S} = MOI.ConstraintIndex{F,S}
 const LBREF = MOICON{MOI.SingleVariable,MOI.GreaterThan{Float64}}
```
Comment (mlubin) [inline src/JuMP.jl]: LBINDEX or MOILB

**Comment by blegat** [comment]
Everything has normally been renamed now :)

**Pull request merged**